spec/controllers/posts_controller_spec.rb

```diff
@@ -1,47 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe PostsController, type: :controller do
-  let(:valid_attributes) {
-    skip("Add a hash of attributes valid for your model")
-  }
 
   let(:invalid_attributes) {
-    skip("Add a hash of attributes invalid for your model")
+    { title: '' }
   }
 
-  # This should return the minimal set of values that should be in the session
-  # in order to pass any filters (e.g. authentication) defined in
-  # PostsController. Be sure to keep this updated too.
-  let(:valid_session) { {} }
+  let(:valid_attributes) {
+    FactoryGirl.attributes_for(:post)
+  }
 
   describe "GET #index" do
     it "assigns all posts as @posts" do
-      post = Post.create! valid_attributes
-      get :index, {}, valid_session
-      expect(assigns(:posts)).to eq([post])
-    end
-  end
-
-  describe "GET #show" do
-    it "assigns the requested post as @post" do
-      post = Post.create! valid_attributes
-      get :show, {:id => post.to_param}, valid_session
-      expect(assigns(:post)).to eq(post)
+      valid_post = FactoryGirl.create(:post)
```

Comment:
I'd recommend using rspec's "let" to memoize the "valid_post" variable and remove the repetition.